### PR TITLE
Extend post_fail_hooks to capture journal from specific units

### PR DESCRIPTION
**Pull request by asdil12**
Ticket: https://progress.opensuse.org/issues/72175
Verification: http://kazhua.qa.suse.de/tests/101#step/rabbitmq/172

### CODE REVIEW

File: tests/console/oom.pm [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    select_console 'root-console';
+
+    my $scriptpath = "/home/$username/data/oom.py";
+
+    #send_key 'alt-sysrq-m';
+    assert_script_run "chmod +x $scriptpath";
+    assert_script_run "$scriptpath";
+}
+
+1;
```
Comment (foursixnine) [inline tests/console/oom.pm:38]: I guess this belongs somewhere else?

Comment (asdil12) [reply tests/console/oom.pm]: yes - I removed it

**Review by foursixnine — APPROVED** [review]

**Comment by juadk** [comment]
Hello guys, I just updated my openQA code in my lab and I'm facing to this issue:
http://1a102.qa.suse.de/tests/5585#step/robot_fw/2
any idea?

**Comment by foursixnine** [comment]
@asdil12 Can you do some verification runs to check that other scenarios that touch those base classes won't be broken? (I guess we'll hear soon enough anyway)

@juadk: Yeah, problem is that hacluster not using `consoletest` or `opensusebasetest` as base class, leads to the method not being defined, as a temporary fix, we could provide a PR that removes the offending line

**Comment by ldevulder** [comment]
> @juadk: Yeah, problem is that hacluster not using `consoletest` or `opensusebasetest` as base class, leads to the method not being defined, as a temporary fix, we could provide a PR that removes the offending line

Yes, this package cannot be used as a class, this should be changed but will take (lot of!) time.

Please remove this line from it, as it blocks pretty much all tests! We could also do it if your want.

**Comment by lilyeyes** [comment]
This push introduce one test case failed: https://openqa.suse.de/tests/4856579#step/oscap_setup/2

**Comment by alvarocarvajald** [comment]
From a quicke search I made, I think this is only impacting tests that use `hacluster` and `openscaptest`:

```
# find ./lib -name "*.pm" -print | xargs grep "use base Exporter" | cut -d: -f1 |xargs grep pre_run_hook | cut -d: -f1 |uniq
./lib/hacluster.pm
./lib/openscaptest.pm
```

Just in case a full revert is not an option.

**Comment by lilyeyes** [comment]
> From a quick search I made, I think this is only impacting tests that use `hacluster` and `openscaptest`:
> 
> ```
> # find ./lib -name "*.pm" -print | xargs grep "use base Exporter" | cut -d: -f1 |xargs grep pre_run_hook | cut -d: -f1 |uniq
> ./lib/hacluster.pm
> ./lib/openscaptest.pm
> ```
> 
> Just in case a full revert is not an option.

Thanks for the info, do you plan to file a poo to fix this "./lib/hacluster.pm ./lib/openscaptest.pm" issue?

**Comment by foursixnine** [comment]
@lilyeyes See #11247

**Comment by foursixnine** [comment]
@lilyeyes :) You're welcome

**Pull request merged**